NewsApp/app/src/main/java/com/thomasvondollen/newsapp/MainActivity.kt

```diff
@@ -66,45 +18,72 @@ class MainActivity : AppCompatActivity() {
         val view = binding.root
         setContentView(view)
 
-        //looping through Article objects and assigning contents to TextViews
-        var count = 0
-
-        for(item in binding.mainGroup.children) {
+        val newsService = NewsService()
```

Comment:
nit: you could move `newsService` up and out of `onCreate()` and set it as a `private val`. This way you'll be able to reference the service from anywhere in the activity.